tests/ha/multi_pod_failure/test_multi_data_pod_failure.py

```diff
@@ -1223,70 +1242,42 @@ def test_continuous_reads_during_kpods_down(self):
-            pod_data.append(resp[2])  # deployment_name
-            self.restore_pod = self.deploy = True
-            self.restore_method = const.RESTORE_DEPLOYMENT_K8S
+            pod_data.append(resp[1][pod_name]['hostname'])  # hostname
+            pod_data.append(resp[1][pod_name]['deployment_backup'])  # deployment_backup
+            pod_data.append(resp[1][pod_name]['deployment_name'])  # deployment_name
+            self.pod_name_list.append(pod_name)
             self.pod_dict[pod_name] = pod_data
-            LOGGER.info("Deleted %s data pod %s by deleting deployment (unsafe)", count, pod_name)
-        LOGGER.info("Step 3: Successfully shutdown %s (K) data pods one by one during "
-                    "READs/VerifyDI on written data in background", self.kvalue)
-        event.clear()
-
-        LOGGER.info("Step 4: Check cluster status")
-        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
-        assert_utils.assert_false(resp[0], resp)
-        LOGGER.info("Step 4: Cluster is in degraded state")
-
-        LOGGER.info("Step 5: Check services status that were running on pods which are deleted.")
-        counter = 0
-        for pod_name in self.pod_name_list:
-            hostname = self.pod_dict.get(pod_name)[0]
-            resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=[pod_name], fail=True,
-                                                               hostname=hostname)
-            LOGGER.debug("Services status on %s : %s", pod_name, resp)
-            if not resp[0]:
-                counter += 1
-            pod_list.remove(pod_name)
-        assert_utils.assert_equal(counter, 0, "Services on some pods not stopped.")
-        LOGGER.info("Step 5: Services of pods are in offline state")
-
-        LOGGER.info("Step 6: Check services status on remaining pods %s", pod_list)
-        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=pod_list, fail=False)
-        LOGGER.debug("Response: %s", resp)
-        assert_utils.assert_true(resp[0], resp)
-        LOGGER.info("Step 6: Services of remaining pods are in online state")
+        self.restore_pod = self.deploy = True
+        self.restore_method = const.RESTORE_DEPLOYMENT_K8S
+        assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
+        LOGGER.info("Step 3: Successfully shutdown %s (K) data pod %s. Verified cluster and "
+                    "services states are as expected & remaining pods status is online.",
+                    self.kvalue, self.pod_name_list)
 
-        LOGGER.info("Step 7: Verify status for In-flight READs/Verify DI while %s (K) pods going "
-                    "down should be failed/error.", self.kvalue)
+        LOGGER.info("Step 4: Verify status for In-flight READs/Verify DI while %s (K) data pods "
+                    "going down should be failed/error.", self.kvalue)
         thread.join()
-        responses = dict()
+        responses = {}
```

Comment:
Why is this reverted?

Reply:
> Why is this reverted?

WIll take care of this change in next PR. ANywasy use of dict{} is giving pylint warning

Reply:
But we decided to declare data structs properly. We can ignore pylint warning as Sarang said.